sentinel-ai-core/src/main/java/com/phonepe/sentinelai/core/earlytermination/EarlyTerminationHandler.java

```diff
@@ -0,0 +1,11 @@
+package com.phonepe.sentinelai.core.earlytermination;
+
+import com.phonepe.sentinelai.core.agent.AgentRunContext;
+import com.phonepe.sentinelai.core.model.ModelRunContext;
+import com.phonepe.sentinelai.core.model.ModelSettings;
+
+@FunctionalInterface
+public interface EarlyTerminationHandler {
+
+    boolean shouldTerminateEarly(ModelSettings modelSettings, ModelRunContext modelRunContext);
```

Comment:
Make this return an enum:
- CONTINUE
- STOP
- ERROR

Usage may not be evident _right_ now, but might come in handy later